packages/material-ui/src/Tabs/ScrollbarSize.js

```diff
@@ -18,53 +17,44 @@ const styles = {
  * The component is originates from https://github.com/STORIS/react-scrollbar-size.
  * It has been moved into the core in order to minimize the bundle size.
  */
-class ScrollbarSize extends React.Component {
-  constructor() {
-    super();
+function ScrollbarSize(props) {
+  const { onChange } = props;
+  const scrollbarHeight = React.useRef();
+  const nodeRef = React.useRef();
 
-    if (typeof window !== 'undefined') {
-      this.handleResize = debounce(() => {
-        const prevHeight = this.scrollbarHeight;
-        this.setMeasurements();
-
-        if (prevHeight !== this.scrollbarHeight) {
-          this.props.onChange(this.scrollbarHeight);
-        }
-      }, 166); // Corresponds to 10 frames at 60 Hz.
+  const setMeasurements = () => {
+    if (!nodeRef.current) {
+      return;
     }
-  }
 
-  componentDidMount() {
-    this.setMeasurements();
-    this.props.onChange(this.scrollbarHeight);
-  }
+    scrollbarHeight.current = nodeRef.current.offsetHeight - nodeRef.current.clientHeight;
+  };
 
-  componentWillUnmount() {
-    this.handleResize.clear();
-  }
+  React.useEffect(() => {
+    const handleResize = debounce(() => {
+      const prevHeight = scrollbarHeight.current;
+      setMeasurements();
 
-  handleRef = ref => {
-    this.nodeRef = ref;
-  };
+      if (prevHeight !== scrollbarHeight.current) {
+        onChange(scrollbarHeight.current);
+      }
+    }, 166); // Corresponds to 10 frames at 60 Hz.
 
-  setMeasurements = () => {
-    const nodeRef = this.nodeRef;
+    window.addEventListener('resize', handleResize);
 
-    if (!nodeRef) {
-      return;
-    }
+    return () => window.removeEventListener('resize', handleResize);
+  }, [onChange]);
 
-    this.scrollbarHeight = nodeRef.offsetHeight - nodeRef.clientHeight;
+  const handleRef = ref => {
```

Comment:
This is likely some redundancy that I missed when removing `findDOMNode`. The ref callback is unnecessary. Just pass along the ref object.